### PR TITLE
Add error handling and staticcheck

### PR DESCRIPTION
**Pull request by yuxiaobo96**
Signed-off-by: yuxiaobo <yuxiaobogo@163.com>

**What type of PR is this?**
> /kind bug

**What this PR does / why we need it**:
Add error handling and staticcheck.
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/master/master.go [hunk 1]
```diff
@@ -419,14 +419,16 @@ func (m *Master) InstallLegacyAPI(c *completedConfig, restOptionsGetter generic.
...
+		return fmt.Errorf("error in registering group versions: %v", err)
 	}
 	return nil
 }
 
 func (m *Master) installTunneler(nodeTunneler tunneler.Tunneler, nodeClient corev1client.NodeInterface) {
 	nodeTunneler.Run(nodeAddressProvider{nodeClient}.externalAddresses)
-	m.GenericAPIServer.AddHealthChecks(healthz.NamedCheck("SSH Tunnel Check", tunneler.TunnelSyncHealthChecker(nodeTunneler)))
+	if err := m.GenericAPIServer.AddHealthChecks(healthz.NamedCheck("SSH Tunnel Check", tunneler.TunnelSyncHealthChecker(nodeTunneler))); err != nil {
+		return
```
Comment (neolit123) [inline pkg/master/master.go]: not calling `klog.Fatalf(..)` here will preserve backwards compatibility but better add a `klog.Warning(...)` (or Warningf) indicating that the call failed?

Comment (yuxiaobo96) [reply pkg/master/master.go]: Ok.
THX.

**Comment by yuxiaobo96** [comment]
/assign @neolit123 

If there is no problem with this pr, please help me approve.
THX.

**Comment by neolit123** [comment]
@yuxiaobo96 
this PR can be approved by sig-apimachinery. @cheftako is assigned.
thanks.

File: pkg/master/master.go [hunk 2]
```diff
@@ -419,14 +419,17 @@ func (m *Master) InstallLegacyAPI(c *completedConfig, restOptionsGetter generic.
...
+		return fmt.Errorf("error in registering group versions: %v", err)
 	}
 	return nil
 }
 
 func (m *Master) installTunneler(nodeTunneler tunneler.Tunneler, nodeClient corev1client.NodeInterface) {
 	nodeTunneler.Run(nodeAddressProvider{nodeClient}.externalAddresses)
-	m.GenericAPIServer.AddHealthChecks(healthz.NamedCheck("SSH Tunnel Check", tunneler.TunnelSyncHealthChecker(nodeTunneler)))
+	if err := m.GenericAPIServer.AddHealthChecks(healthz.NamedCheck("SSH Tunnel Check", tunneler.TunnelSyncHealthChecker(nodeTunneler))); err != nil {
+		klog.Warning("error adding health check for SSH tunnel")
```
Comment (cheftako) [inline pkg/master/master.go]: It would be nice to know what the error is. Can we include it in the warning?

**Comment by yuxiaobo96** [comment]
This pr has been on hold for too long, so close.

**Pull request closed without merging**